mdr/mdr.py

```diff
@@ -57,6 +60,22 @@ def fit(self, features, classes):
         None
 
         """
+        n = (np.unique(classes)).size # general case 
+        for i in range(features.shape[0]):
+            feature_instance = tuple(map(tuple, features[i])) 
+            if feature_instance not in dict: 
+                feature_map[feature_instance] = np.zeros((n,), dtype=np.int) #initialize count for a new set of feature values 
```

Comment:
Use a `defaultdict` to avoid having to check whether a feature instance is in the dictionary.